Source/SwiftLintFramework/Rules/Idiomatic/DiscouragedNoneName.swift

```diff
@@ -0,0 +1,217 @@
+import SourceKittenFramework
+
+public struct DiscouragedNoneName: ASTRule, OptInRule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+    public static var description = RuleDescription(
+        identifier: "discouraged_none_name",
+        name: "Discouraged None Name",
+        description: "Discourages name cases/static members `none`, which can conflict with `Optional<T>.none`.",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            // Should not trigger unless exactly matches "none"
+            Example("""
+            enum MyEnum {
+                case nOne
+            }
+            """),
+            Example("""
+            enum MyEnum {
+                case _none
+            }
+            """),
+            Example("""
+            enum MyEnum {
+                case none_
+            }
+            """),
+            Example("""
+            enum MyEnum {
+                case none(Any)
+            }
+            """),
+            Example("""
+            enum MyEnum {
+                case nonenone
+            }
+            """),
+            Example("""
+            class MyClass {
+                class var nonenone: MyClass { MyClass() }
+            }
+            """),
+            Example("""
+            class MyClass {
+                static var nonenone = MyClass()
+            }
+            """),
+            Example("""
+            class MyClass {
+                static let nonenone = MyClass()
+            }
+            """),
+            Example("""
+            struct MyStruct {
+                static var nonenone = MyStruct()
+            }
+            """),
+            Example("""
+            struct MyStruct {
+                static let nonenone = MyStruct()
+            }
+            """),
+
+            // Should not trigger if not an enum case or static/class member
+            Example("""
+            struct MyStruct {
+                let none = MyStruct()
+            }
+            """),
+            Example("""
+            struct MyStruct {
+                var none = MyStruct()
+            }
+            """),
+            Example("""
+            class MyClass {
+                let none = MyClass()
+            }
+            """),
+            Example("""
+            class MyClass {
+                var none = MyClass()
+            }
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            enum MyEnum {
```

Comment:
can you please update the triggering examples with the violation marker (↓) so that the unit tests can also validate where the violation is triggered?

Reply:
Yep of course! I did wonder what the arrow was for, it makes sense now. Updated, thanks!